src/server/editorServices.ts

```diff
@@ -463,7 +448,45 @@ namespace ts.server {
                     this.typingsCache.deleteTypingsForProject(response.projectName);
                     break;
             }
-            project.updateGraph();
+            project.markAsDirty();
+            this.delayUpdateProjectGraphAndInferredProjectsRefresh(project);
+        }
+
+        private delayInferredProjectsRefresh() {
+            this.pendingInferredProjectUpdate = true;
+            this.throttledOperations.schedule("*refreshInferredProjects*", /*delay*/ 250, () => {
+                if (this.pendingProjectUpdates.size !== 0) {
+                    this.delayInferredProjectsRefresh();
+                }
+                else if (this.pendingInferredProjectUpdate) {
+                    this.pendingInferredProjectUpdate = false;
+                    this.refreshInferredProjects();
+                }
+            });
+        }
+
+        private delayUpdateProjectGraph(project: Project) {
+            const projectName = project.getProjectName();
+            this.pendingProjectUpdates.set(projectName, project);
+            this.throttledOperations.schedule(projectName, /*delay*/ 250, () => {
+                const project = this.pendingProjectUpdates.get(projectName);

```

Comment:
I think i like the current code just because it avoids creating unnecessary closure